smoke-tests/framework/src/main/java/com/microsoft/applicationinsights/smoketest/DependencyAssert.java

```diff
@@ -62,7 +62,29 @@ public DependencyAssert hasNoParent() {
   @CanIgnoreReturnValue
   public DependencyAssert hasParent(String parentId) {
     isNotNull();
-    assertThat(getDependencyData().getProperties().get("ai.operation.parentId")).isNull();
+    assertThat(getDependencyData().getProperties().get("ai.operation.parentId"))
+        .isEqualTo(parentId);
```

Comment:
also fix hasNoParent above

```suggestion
  hasTag("ai.operation.parentId", parentId);
```